pkg/api/domains/domains_test.go

```diff
@@ -0,0 +1,273 @@
+	Domains = nil
+	clusters := testutil.GetTestClusters()
+	var clients kubernetes.Interface = fake.NewSimpleClientset()
+	modelBindingPairs := map[string]*types.ModelBindingPair{
+		"test-pair-1": getModelBindingPair(),
+	}
+
+	tests := []struct {
+		name    string
+		listers controller.Listers
+	}{
+		{name: "testReturn404ForMissingDomain",
+			listers: testutilcontroller.NewControllerListers(&clients, clusters, &modelBindingPairs)},
+	}
+	// run the tests
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// GIVEN an array of clusters and a model binding pair that reference a WLS domain
+			createAndInsertWLSDomain(tt)
+			// setup http request processing
+			vars := map[string]string{
+				"id": "non-existent",
+			}
+			response := httptest.NewRecorder()
+			request := httptest.NewRequest("GET", "/domains", nil)
+			request = mux.SetURLVars(request, vars)
+			// WHEN a request is submitted for a non-existent domain
+			ReturnSingleDomain(response, request)
+			// THEN a NOT FOUND HTTP status response message will be returned
+			assert.Equal(t, http.StatusNotFound, response.Code, "wrong response code")
+		})
+	}
+}
+
+// TestReturnSingleDomain will test that a configured domain is returned
+func TestReturnSingleDomain(t *testing.T) {
+	Domains = nil
+	clusters := testutil.GetTestClusters()
+	var clients kubernetes.Interface = fake.NewSimpleClientset()
+	modelBindingPairs := map[string]*types.ModelBindingPair{
+		"test-pair-1": getModelBindingPair(),
+	}
+
+	tests := []struct {
+		name    string
+		listers controller.Listers
+	}{
+		{name: "testReturnSingleDomain",
+			listers: testutilcontroller.NewControllerListers(&clients, clusters, &modelBindingPairs)},
+	}
+	// run the tests
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// GIVEN an array of clusters and a model binding pair referencing a WLS domain
+			createAndInsertWLSDomain(tt)
+			// setup http request processing
+			vars := map[string]string{
+				"id": "test-weblogic",
+			}
+			response := httptest.NewRecorder()
+			request := httptest.NewRequest("GET", "/domains", nil)
+			request = mux.SetURLVars(request, vars)
+			// WHEN a request is submitted for domain with ID "test-weblogic"
+			ReturnSingleDomain(response, request)
+			// THEN the response will has a HTTP OK status (200) with a single domain JSON response body
+			assert.Equal(t, http.StatusOK, response.Code, "wrong response code")
+			var domain Domain
+			err := json.Unmarshal(response.Body.Bytes(), &domain)
+			if err != nil {
+				t.Errorf("Error while reading response JSON: %s", err)
+			}
+			// Validate the correct number of domains (1), and the domain has the expected namespace
+			assert.NotNil(t, domain, "Domain should not be nil")
+			assert.Equal(t, "test3", domain.Namespace, "Domain has the wrong namespace")
+		})
+	}
+}
+
+//createAndInsertWLSDomain will augment the clusters referenced by the listers with a WLSDomain
+func createAndInsertWLSDomain(tt struct {
```

Comment:
I have never seen a struct defined in the signature like this, don't think it is common.  Can't you just pass in tt.listers?

Reply:
OK. I'll try to address it.  But if it passes all the formatting and lint checks, wouldn't it be considered valid?

Reply:
the issue here is that the structure is defined internally in the calling methods, so it really needs to be defined for the target method as well.  I'd rather maintain things as is given the unit test structure is following a format that we discusses as a team and agreed is valid.

Reply:
what i am saying is that I think you do not need to pass the entire struct, just pass in tt.listers.  The function doesn't use tt.name does it?

Reply:
done